src/managers/XWaylandManager.cpp

```diff
@@ -319,27 +317,15 @@ Vector2D CHyprXWaylandManager::getMaxSizeForWindow(CWindow* pWindow) {
 
 Vector2D CHyprXWaylandManager::xwaylandToWaylandCoords(const Vector2D& coord) {
 
-    static auto* const PXWLFORCESCALEZERO = &g_pConfigManager->getConfigValuePtr("xwayland:force_zero_scaling")->intValue;
-
-    CMonitor*          pMonitor     = nullptr;
-    double             bestDistance = __FLT_MAX__;
-    for (auto& m : g_pCompositor->m_vMonitors) {
-        const auto SIZ = *PXWLFORCESCALEZERO ? m->vecTransformedSize : m->vecSize;
-
-        double     distance =
-            vecToRectDistanceSquared(coord, {m->vecXWaylandPosition.x, m->vecXWaylandPosition.y}, {m->vecXWaylandPosition.x + SIZ.x - 1, m->vecXWaylandPosition.y + SIZ.y - 1});
-
-        if (distance < bestDistance) {
-            bestDistance = distance;
-            pMonitor     = m.get();
-        }
-    }
+    CMonitor* pMonitor = g_pCompositor->getMonitorFromVector(coord);
```

Comment:
you can't do that

Reply:
2.0 - double, 2.f - float. The compiler will complain in strict mode. But the reason is different, when I wrote the code I used 2.0 in many places, then I corrected everything together.

Reply:
@vaxerski I looked at the code and it's identical.

Reply:
well then you looked wrong.